src/datatable/datatable.jl

```diff
@@ -764,8 +764,16 @@ hcat!(dt::DataTable, x) = hcat!(dt, DataTable(Any[NullableArray([x])]))
 hcat!(dt::DataTable) = dt
 hcat!(a::DataTable, b, c...) = hcat!(hcat!(a, b), c...)
 
-# hcat
+# hcat for 2 arguments
 Base.hcat(dt::DataTable, x) = hcat!(copy(dt), x)
+Base.hcat(dt1::DataTable, dt2::AbstractDataTable) = hcat!(copy(dt1), dt2)
+
+# hcat! for >2 arguments
```

Comment:
I don't think this comment is really needed, it's quite clear from the signature already.